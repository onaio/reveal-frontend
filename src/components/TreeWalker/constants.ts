export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
