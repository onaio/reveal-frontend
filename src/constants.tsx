// magical strings
export const JURISDICTION_ID = 'jurisdiction_id';
export const JURISDICTIONID = 'jurisdictionId';
export const PARENT_ID = 'parent_id';
export const PARENTID = 'parentId';
export const CASE_TRIGGERED = 'Case Triggered';
export const ROUTINE = 'Routine';
export const INVESTIGATION = 'Investigation';
export const STRUCTURE_LAYER = 'structure-layer';
export const POLYGON = 'Polygon';
export const MULTI_POLYGON = 'MultiPolygon';
export const POINT = 'Point';
export const FEATURE = 'Feature';
export const FEATURE_COLLECTION = 'FeatureCollection';
export const MAIN_PLAN = 'main-plan-layer';
export const APP = 'APP';
export const MAP_ID = 'map-1';
export const GEOJSON = 'geojson';
export const STRINGIFIED_GEOJSON = 'stringified-geojson';
export const MAP = 'map';
export const MAP_AREA = 'map-area';
export const DRAFT = 'draft';
export const NEW = 'new';
export const A1 = 'A1';
export const A2 = 'A2';
export const B1 = 'B1';
export const B2 = 'B2';
export const LOW_PRIORITY = 'low-priority';
export const MEDIUM_PRIORITY = 'medium-priority';
export const HIGH_PRIORITIY = 'high-priority';
export const TRUE = 'True';
export const FALSE = 'False';
export const DISABLED = 'Disabled';
export const INTERVENTION_TYPE_CODE = 'interventionType';
export const FI_STATUS_CODE = 'fiStatus';
export const FI_REASON_CODE = 'fiReason';
export const OPENSRP_EVENT_ID_CODE = 'opensrpEventId';
export const CASE_NUMBER_CODE = 'caseNum';
export const TASK_GENERATION_STATUS_CODE = 'taskGenerationStatus';
export const GA_ENV_TEST = 'test';
export const PLAN_ID = 'plan_id';
export const PLAN_INTERVENTION_TYPE = 'plan_intervention_type';
export const TWO_HUNDRED_PX = '200px';
export const PLAN_RECORD_BY_ID = 'planRecordsById';
export const MAPBOXGL_POPUP = '.mapboxgl-popup';
export const TABLE_BORDERED_CLASS = 'table table-bordered';
export const ACTION_CODE = 'action_code';
export const CASE_CONFIRMATION_GOAL_ID = 'Case_Confirmation';
export const MOSQUITO_COLLECTION_ID = 'Mosquito_Collection';
export const LARVAL_DIPPING_ID = 'Larval_Dipping';
export const RACD_REGISTER_FAMILY_ID = 'RACD_register_families';
export const STRUCTURES_LINE = 'structures-line';
export const STRUCTURES_FILL = 'structures-fill';
export const HISTORICAL_INDEX_CASES = 'historical-index-cases';
export const CURRENT_INDEX_CASES = 'current-index-cases';
export const FILE_UPLOAD_TYPE = 'file-upload';
export const VALIDATOR_UPLOAD_TYPE = 'validator-upload';
export const DAYS = 'days';
export const CONDITION = 'condition';
export const TRIGGER = 'trigger';
export const RISK_LABEL = 'Risk';
export const TIMELINE_SLIDER_STEP1 = '1';
export const TIMELINE_SLIDER_STEP2 = '2';
export const TIMELINE_SLIDER_STEP3 = '3';
export const MODAL_BUTTON_CLASS = 'focus-investigation btn btn-primary float-right mt-0';

// internal urls
export const BACKEND_LOGIN_URL = '/fe/login';
export const EXPRESS_LOGIN_URL = '/login';
export const REACT_LOGIN_URL = '/login';
export const LOGOUT_URL = '/logout';
export const HOME_URL = '/';
export const REPORT_IRS_PLAN_URL = `/intervention/irs/report`;
export const REPORT_MDA_POINT_PLAN_URL = `/intervention/mda-point/report`;
export const REPORT_MDA_PLAN_URL = `/intervention/mda/report`;
export const MDA_POINT_LOCATION_REPORT_URL = '/intervention/mda-point/location-report';
export const MDA_POINT_CHILD_REPORT_URL = '/intervention/mda-point/child-report';
export const ASSIGN_PLAN_URL = `/assign`;
export const FI_URL = '/focus-investigation';
export const FI_FILTER_URL = '/focus-investigation/filter';
export const FI_SINGLE_URL = '/focus-investigation/view';
export const FI_SINGLE_MAP_URL = '/focus-investigation/map';
export const NEW_PLAN_URL = '/plans/new';
export const PLAN_LIST_URL = '/plans/list';
export const PLAN_COMPLETION_URL = '/focus-investigation/view/complete';
export const PLAN_UPDATE_URL = '/plans/update';
export const CREATE_ORGANIZATION_URL = '/teams/new';
export const EDIT_ORGANIZATION_URL = '/teams/edit';
export const SINGLE_ORGANIZATION_URL = '/teams/view';
export const ORGANIZATIONS_LIST_URL = '/teams';
export const CREATE_PRACTITIONER_URL = '/practitioners/new';
export const EDIT_PRACTITIONER_URL = '/practitioners/edit';
export const ASSIGN_ORGANIZATION_URL = '/plans/teamAssignment';
export const PRACTITIONERS_LIST_URL = '/practitioners';
export const ASSIGN_PRACTITIONERS_URL = '/teams/assignPractitioners';
export const PLAN = 'plan';
export const REPORT = 'report';
export const BACKEND_CALLBACK_URL = '/fe/oauth/callback/opensrp';
export const BACKEND_CALLBACK_PATH = '/fe/oauth/callback/:id';
export const REACT_CALLBACK_PATH = '/oauth/callback/:id';
export const CLIENTS_LIST_URL = '/clients';
export const UPLOAD_CLIENT_CSV_URL = '/clients/upload';
export const JURISDICTION_METADATA_URL = '/jurisdiction-metadata';
export const GO_BACK_TEXT = 'Go Back';
export const MANIFEST_RELEASE_URL = '/manifest/releases';
export const VIEW_DRAFT_FILES_URL = '/files/draft';
export const JSON_VALIDATORS_URL = '/json-validators';
export const MANIFEST_FILE_UPLOAD = '/manifest';
export const EDIT_SERVER_SETTINGS_URL = '/server-settings';
export const PLANNING_VIEW_URL = '/plans/planning';
export const NEW_PLANNING_PLAN_URL = `${PLANNING_VIEW_URL}/${NEW}`;
export const ASSIGN_JURISDICTIONS_URL = '/assignJurisdictions';
export const MANUAL_ASSIGN_JURISDICTIONS_URL = '/manualSelectJurisdictions';
export const AUTO_ASSIGN_JURISDICTIONS_URL = '/auoSelectJurisdictions';
/** how long after opening a logout window should we wait before redirecting to express' server logout
 * I am not sure the optimum value for this, mozilla firefox seems to take some
 * time loading up the logoutWindow and actually making the request.
 */
export const LOGOUT_REDIRECTION_DELAY = 1000;

// OpenSRP API strings
export const OPENSRP_PRACTITIONER_ENDPOINT = 'practitioner';
export const OPENSRP_PRACTITIONER_ROLE_ENDPOINT = 'practitionerRole';
export const OPENSRP_FIND_BY_PROPERTIES = 'findByProperties';
export const OPENSRP_LOCATION = 'location';
export const OPENSRP_PLANS = 'plans';
export const OPENSRP_PARENT_ID = 'parent_id';
export const OPENSRP_ORGANIZATION_ENDPOINT = 'organization';
export const OPENSRP_GET_ASSIGNMENTS_ENDPOINT = `${OPENSRP_ORGANIZATION_ENDPOINT}/assignedLocationsAndPlans`;
export const OPENSRP_POST_ASSIGNMENTS_ENDPOINT = `${OPENSRP_ORGANIZATION_ENDPOINT}/assignLocationsAndPlans`;
export const OPENSRP_ORG_PRACTITIONER_ENDPOINT = 'organization/practitioner';
export const OPENSRP_DEL_PRACTITIONER_ROLE_ENDPOINT = 'practitionerRole/deleteByPractitioner';
export const OPENSRP_ADD_PRACTITIONER_ROLE_ENDPOINT = 'practitionerRole/add';
export const OPENSRP_USERS_ENDPOINT = 'user';
export const OPENSRP_USERS_COUNT_ENDPOINT = 'user/count';
export const OPENSRP_FIND_EVENTS_ENDPOINT = 'event/findById';
export const OPENSRP_LOCATIONS_BY_PLAN = 'plans/findLocationNames';
export const OPENSRP_FILE_UPLOAD_HISTORY_ENDPOINT = 'upload/history';
export const OPENSRP_EVENT_PARAM_VALUE = 'Child Registration';
export const OPENSRP_UPLOAD_ENDPOINT = 'upload';
export const OPENSRP_UPLOAD_DOWNLOAD_ENDPOINT = 'upload/download';
export const EVENT_NAME_PARAM = 'event_name';
export const LOCATION_ID_PARAM = 'location_id';
export const TEAM_ID_PARAM = 'team_id';
export const OPENSRP_TEMPLATE_ENDPOINT = 'template';
export const OPENSRP_PLANS_BY_USER_FILTER = 'plans/user';
export const OPENSRP_V1_SETTINGS_ENDPOINT = 'settings/sync';
export const OPENSRP_V2_SETTINGS = 'v2/settings/';
export const OPENSRP_JURISDICTION_HIERARCHY_ENDPOINT = 'location/hierarchy';
export const OPENSRP_PLAN_HIERARCHY_ENDPOINT = 'location/hierarchy/plan';
export const OPENSRP_FIND_LOCATION_BY_JURISDICTION_IDS = `location/findByJurisdictionIds`;
export const OPENSRP_ACTIVE = 'Active';
export const OPENSRP_STATUS = 'status';
export const OPENSRP_MANIFEST_ENDPOINT = 'manifest';
export const OPENSRP_FORMS_ENDPOINT = 'clientForm';
export const OPENSRP_MANIFEST_FORMS_ENDPOINT = 'clientForm/release-related-files';
export const OPENSRP_FORM_METADATA_ENDPOINT = 'clientForm/metadata';
export const SETTINGS_ENDPOINT = 'settings/';
export const LOCATIONS_ENDPOINT = 'location/location-tree';
export const SECURITY_AUTHENTICATE_ENDPOINT = 'security/authenticate';
export const OPENSRP_KEYCLOAK_PARAM = 'Keycloak';

// colors
export const GREEN = 'Green';
export const YELLOW = 'Yellow';
export const ORANGE = 'Orange';
export const RED = 'Red';

// plan activity codes
export const CASE_CONFIRMATION_ACTIVITY_CODE = 'caseConfirmation';
export const FAMILY_REGISTRATION_ACTIVITY_CODE = 'familyRegistration';
export const BLOOD_SCREENING_ACTIVITY_CODE = 'bloodScreening';
export const BEDNET_DISTRIBUTION_ACTIVITY_CODE = 'bednetDistribution';
export const LARVAL_DIPPING_ACTIVITY_CODE = 'larvalDipping';
export const MOSQUITO_COLLECTION_ACTIVITY_CODE = 'mosquitoCollection';
export const BCC_ACTIVITY_CODE = 'BCC';
export const IRS_ACTIVITY_CODE = 'IRS';
export const MDA_POINT_DISPENSE_ACTIVITY_CODE = 'pointDispenseMDA';
export const MDA_POINT_ADVERSE_EFFECTS_ACTIVITY_CODE = 'pointAdverseMDA';
export const DYNAMIC_FAMILY_REGISTRATION_ACTIVITY_CODE = 'dynamicFamilyRegistration';
export const DYNAMIC_BLOOD_SCREENING_ACTIVITY_CODE = 'dynamicBloodScreening';
export const DYNAMIC_BEDNET_DISTRIBUTION_ACTIVITY_CODE = 'dynamicBednetDistribution';
export const DYNAMIC_LARVAL_DIPPING_ACTIVITY_CODE = 'dynamicLarvalDipping';
export const DYNAMIC_MOSQUITO_COLLECTION_ACTIVITY_CODE = 'dynamicMosquitoCollection';
export const DYNAMIC_BCC_ACTIVITY_CODE = 'dynamicBCC';
export const DYNAMIC_IRS_ACTIVITY_CODE = 'dynamicIRS';
export const DYNAMIC_MDA_COMMUNITY_DISPENSE_ACTIVITY_CODE = 'dynamicCommunityDispenseMDA';
export const DYNAMIC_MDA_COMMUNITY_ADHERENCE_ACTIVITY_CODE = 'dynamicCommunityAdherenceMDA';

// task action codes
export const BCC_CODE = 'BCC';
export const IRS_CODE = 'IRS';
export const BEDNET_DISTRIBUTION_CODE = 'Bednet Distribution';
export const BLOOD_SCREENING_CODE = 'Blood Screening';
export const CASE_CONFIRMATION_CODE = 'Case Confirmation';
export const RACD_REGISTER_FAMILY_CODE = 'RACD Register Family';
export const LARVAL_DIPPING_CODE = 'Larval Dipping';
export const MOSQUITO_COLLECTION_CODE = 'Mosquito Collection';
export const MDA_POINT_DISPENSE_CODE = 'MDA Dispense';
export const MDA_POINT_ADVERSE_EFFECTS_CODE = 'MDA Adverse Event(s)';
export const MDA_ADHERENCE_CODE = 'MDA Adherence';
export const GOAL_CONFIRMATION_GOAL_ID = 'Case_Confirmation';

export const CREATE_TYPE = 'create';

// dynamic plan activities
export const NAMED_EVENT_TRIGGER_TYPE = 'named-event';
export const PLAN_ACTIVATION_TRIGGER_NAME = 'plan-activation';
export const APPLICABILITY_CONDITION_KIND = 'applicability';

export const PRACTITIONER_CODE = {
  text: 'Community Health Worker',
};
/** Field to sort plans by */
export const SORT_BY_EFFECTIVE_PERIOD_START_FIELD = 'plan_effective_period_start';

/** Query Params */
export const QUERY_PARAM_TITLE = 'title';
export const QUERY_PARAM_USER = 'user';
export const REACTIVE_QUERY_PARAM = 'reactive_plans';
export const ROUTINE_QUERY_PARAM = 'routine_plans';

/** Settings Configuration */
export const SETTINGS_CONFIGURATION = 'SettingConfiguration';

/** Jurisdiction Metadata */
export const JURISDICTION_METADATA_RISK = 'jurisdiction_metadata-risk';
export const JURISDICTION_METADATA_COVERAGE = 'jurisdiction_metadata-coverage';
export const JURISDICTION_METADATA_POPULATION = 'jurisdiction_metadata-population';
export const JURISDICTION_METADATA_TARGET = 'jurisdiction_metadata-target';
export const JURISDICTION_CSV_TEMPLATE = 'jurisdiction_id,jurisdiction_name,risk,coverage';
export const JURISDICTION_CSV_FILE_NAME = 'jurisdiction-metadata.csv';
export const GET_ALL = 'getAll';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';

/** Superset API strings */
export const SUPERSET_ACCESS_DENIED_MESSAGE = 'Access is Denied';

/** React Mapbox GL strings */
export const REACT_MAPBOX_GL_ICON_IMAGE = 'icon-image';
export const REACT_MAPBOX_GL_ICON_SIZE = 'icon-size';
