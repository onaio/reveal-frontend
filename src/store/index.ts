import { store } from '@opensrp/store';

export default store;
