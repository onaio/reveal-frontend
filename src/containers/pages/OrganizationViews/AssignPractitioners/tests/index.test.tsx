import reducerRegistry from '@onaio/redux-reducer-registry';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import Helmet from 'react-helmet';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import ConnectedAssignPractitioner, { AssignPractitioner } from '..';
import { ASSIGN, PRACTITIONERS } from '../../../../../configs/lang';
import { ASSIGN_PRACTITIONERS_URL, FI_URL } from '../../../../../constants';
import { OpenSRPService } from '../../../../../services/opensrp';
import store from '../../../../../store';
import organizationsReducer, {
  fetchOrganizations,
  reducerName as organizationReducerName,
  removeOrganizationsAction,
} from '../../../../../store/ducks/opensrp/organizations';
import practitionersReducer, {
  reducerName as practitionerReducerName,
} from '../../../../../store/ducks/opensrp/practitioners';
import { removePractitionersAction } from '../../../../../store/ducks/opensrp/practitioners';
import * as fixtures from '../../../../../store/ducks/tests/fixtures';

reducerRegistry.register(organizationReducerName, organizationsReducer);
reducerRegistry.register(practitionerReducerName, practitionersReducer);

// tslint:disable-next-line: no-var-requires
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

describe('src/pages/*/AssignPractitioners', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    store.dispatch(removePractitionersAction);
    store.dispatch(removeOrganizationsAction);
  });

  it('render without crashing', () => {
    fetch
      .once(JSON.stringify(fixtures.organization3))
      .once(JSON.stringify(fixtures.org3Practitioners))
      .once(JSON.stringify(fixtures.allPractitioners));

    const mock: any = jest.fn();
    const props = {
      fetchOrganizationsCreator: fetchOrganizations,
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.organization3.identifier },
        path: `${ASSIGN_PRACTITIONERS_URL}/:id`,
        url: `${ASSIGN_PRACTITIONERS_URL}/${fixtures.organization3.identifier}`,
      },
      organization: fixtures.organization3,
      serviceClass: OpenSRPService,
    };

    shallow(
      <Router history={history}>
        <AssignPractitioner {...props} />
      </Router>
    );
  });

  it('Works well with store', async () => {
    // interest here is if we component receives correct props
    fetch
      .once(JSON.stringify(fixtures.organization3))
      .once(JSON.stringify(fixtures.org3Practitioners))
      .once(JSON.stringify(fixtures.allPractitioners));

    const mock: any = jest.fn();
    const props = {
      fetchOrganizationsCreator: fetchOrganizations,
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.organization3.identifier },
        path: `${ASSIGN_PRACTITIONERS_URL}/:id`,
        url: `${ASSIGN_PRACTITIONERS_URL}/${fixtures.organization3.identifier}`,
      },
      serviceClass: OpenSRPService,
    };

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedAssignPractitioner {...props} />
        </Router>
      </Provider>
    );

    await new Promise<any>(resolve => setImmediate(resolve));
    wrapper.update();

    const theProps = wrapper.find('AssignPractitioner').props() as any;
    expect(theProps.organization).toEqual(fixtures.organization3);
    expect(theProps.assignedPractitioners).toEqual(fixtures.org3Practitioners);
  });

  it('Renders correctly', async () => {
    // interest here is if we have crucial page components
    fetch
      .once(JSON.stringify({}))
      .once(JSON.stringify([]))
      .once(JSON.stringify([]));

    const mock: any = jest.fn();
    const props = {
      assignedPractitioners: fixtures.org3Practitioners,
      fetchOrganizationsCreator: fetchOrganizations,
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.organization3.identifier },
        path: `${ASSIGN_PRACTITIONERS_URL}/:id`,
        url: `${ASSIGN_PRACTITIONERS_URL}/${fixtures.organization3.identifier}`,
      },
      organization: fixtures.organization3,
      serviceClass: OpenSRPService,
    };

    const wrapper = mount(
      <Router history={history}>
        <AssignPractitioner {...props} />
      </Router>
    );

    wrapper.update();

    // page Title
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual(`${ASSIGN} ${PRACTITIONERS}`);

    // assigned-options spans
    expect(wrapper.find('span.assigned-options').length).toEqual(3);

    // BreadCrumb
    expect(wrapper.find('Breadcrumb').length).toEqual(1);
    expect(toJson(wrapper.find('Breadcrumb'))).toMatchSnapshot('BreadCrumb');

    // the async select component
    expect(wrapper.find('Select input').length).toEqual(1);
    expect(toJson(wrapper.find('Select input'))).toMatchSnapshot('async select');

    // Select values
    expect(wrapper.find('Select').prop('value')).toMatchSnapshot('Select Values');

    // add button
    expect(toJson(wrapper.find('#add-button'))).toMatchSnapshot('add Practitioners button');

    // discard button
    expect(toJson(wrapper.find('#discard-button'))).toMatchSnapshot('add Practitioners button');
  });

  it('The async select works correctly', async () => {
    // check that you can select one, for nominal case,
    fetch
      .once(JSON.stringify(fixtures.organization3))
      .once(JSON.stringify([]))
      .once(JSON.stringify(fixtures.allPractitioners));

    const mock: any = jest.fn();
    const props = {
      assignedPractitioners: [],
      fetchOrganizationsCreator: fetchOrganizations,
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.organization3.identifier },
        path: `${ASSIGN_PRACTITIONERS_URL}/:id`,
        url: `${ASSIGN_PRACTITIONERS_URL}/${fixtures.organization3.identifier}`,
      },
      organization: fixtures.organization3,
      serviceClass: OpenSRPService,
    };

    const wrapper = mount(
      <Router history={history}>
        <AssignPractitioner {...props} />
      </Router>
    );

    await flushPromises();
    wrapper.update();

    const select = wrapper.find('Select');
    // simulate single value change
    const entry = fixtures.practitioner2;
    (select.instance() as any).selectOption({
      label: `${entry.username} - ${entry.name}`,
      value: entry.identifier,
    });
    wrapper.update();

    expect(wrapper.find('Select').props().value).toEqual([
      {
        label: 'tak - Biophics Tester',
        value: 'd7c9c000-e9b3-427a-890e-49c301aa48e6',
      },
    ]);

    // simulate backspace
    wrapper.find('Select').simulate('keyDown', { key: 'Backspace', keyCode: 8 });
    wrapper.update();

    expect(wrapper.find('Select').props().value).toEqual([]);
  });

  it('Should not navigate when there are unsaved changes', async () => {
    fetch
      .once(JSON.stringify(fixtures.organization3))
      .once(JSON.stringify([]))
      .once(JSON.stringify(fixtures.allPractitioners));

    const mock: any = jest.fn();
    const props = {
      assignedPractitioners: [],
      fetchOrganizationsCreator: fetchOrganizations,
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.organization3.identifier },
        path: `${ASSIGN_PRACTITIONERS_URL}/:id`,
        url: `${ASSIGN_PRACTITIONERS_URL}/${fixtures.organization3.identifier}`,
      },
      organization: fixtures.organization3,
      serviceClass: OpenSRPService,
    };

    const wrapper = mount(
      <Router history={history}>
        <AssignPractitioner {...props} />
      </Router>
    );
    await flushPromises();
    wrapper.update();

    expect((wrapper.find('Prompt') as any).props().when).toBeFalsy();

    const select = wrapper.find('Select');
    // simulate single value change function
    const entry = fixtures.practitioner2;
    (select.instance() as any).selectOption({
      label: `${entry.username} - ${entry.name}`,
      value: entry.identifier,
    });
    wrapper.update();

    history.push(FI_URL);
    wrapper.update();
    const message = `Unsaved Changes: please Save or Discard changes made [${fixtures.organization3.name}]`;
    expect((wrapper.find('Prompt') as any).props().message()).toEqual(message);
    expect((wrapper.find('Prompt') as any).props().when).toBeTruthy();
  });

  it('discard changes works correctly', async () => {
    fetch
      .once(JSON.stringify(fixtures.organization3))
      .once(JSON.stringify([]))
      .once(JSON.stringify(fixtures.allPractitioners));

    const mock: any = jest.fn();
    const props = {
      assignedPractitioners: [],
      fetchOrganizationsCreator: fetchOrganizations,
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.organization3.identifier },
        path: `${ASSIGN_PRACTITIONERS_URL}/:id`,
        url: `${ASSIGN_PRACTITIONERS_URL}/${fixtures.organization3.identifier}`,
      },
      organization: fixtures.organization3,
      serviceClass: OpenSRPService,
    };

    const wrapper = mount(
      <Router history={history}>
        <AssignPractitioner {...props} />
      </Router>
    );

    expect(wrapper.text().includes('tak - Biophics')).toBeFalsy();

    await flushPromises();
    wrapper.update();

    const select = wrapper.find('Select');
    // simulate single value change
    const entry = fixtures.practitioner2;
    (select.instance() as any).selectOption({
      label: `${entry.username} - ${entry.name}`,
      value: entry.identifier,
    });
    wrapper.update();

    expect(wrapper.text().includes('tak - Biophics')).toBeTruthy();
    wrapper.update();

    wrapper.find('button#discard-button').simulate('click');
    wrapper.update();
    // confirm the option is no longer rendered
    expect(wrapper.text().includes('tak - Biophics')).toBeFalsy();

    // check that on clicking discard changes it redirects to team view
    wrapper.find('button#discard-button').simulate('click');
    wrapper.update();
    expect(history.location.pathname).toEqual('/teams/view/d23f7350-d406-11e9-bb65-2a2ae2dbcce4');
  });
});
