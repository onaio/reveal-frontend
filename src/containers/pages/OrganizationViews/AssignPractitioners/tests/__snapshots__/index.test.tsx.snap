// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/*/AssignPractitioners Renders correctly: BreadCrumb 1`] = `
<Breadcrumb
  aria-label="breadcrumb"
  className="reveal-breadcrumb"
  listTag="ol"
  tag="nav"
>
  <nav
    aria-label="breadcrumb"
    className="reveal-breadcrumb"
  >
    <ol
      className="breadcrumb"
    >
      <BreadcrumbItem
        key="reveal-breadcrumb-0"
        tag="li"
      >
        <li
          className="breadcrumb-item"
        >
          <Link
            key="0"
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </LinkAnchor>
          </Link>
        </li>
      </BreadcrumbItem>
      <BreadcrumbItem
        key="reveal-breadcrumb-1"
        tag="li"
      >
        <li
          className="breadcrumb-item"
        >
          <Link
            key="1"
            to="/teams"
          >
            <LinkAnchor
              href="/teams"
              navigate={[Function]}
            >
              <a
                href="/teams"
                onClick={[Function]}
              >
                Teams
              </a>
            </LinkAnchor>
          </Link>
        </li>
      </BreadcrumbItem>
      <BreadcrumbItem
        key="reveal-breadcrumb-2"
        tag="li"
      >
        <li
          className="breadcrumb-item"
        >
          <Link
            key="2"
            to="/teams/view/d23f7350-d406-11e9-bb65-2a2ae2dbcce4"
          >
            <LinkAnchor
              href="/teams/view/d23f7350-d406-11e9-bb65-2a2ae2dbcce4"
              navigate={[Function]}
            >
              <a
                href="/teams/view/d23f7350-d406-11e9-bb65-2a2ae2dbcce4"
                onClick={[Function]}
              >
                Takang 1
              </a>
            </LinkAnchor>
          </Link>
        </li>
      </BreadcrumbItem>
      <BreadcrumbItem
        active={true}
        tag="li"
      >
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Assign Practitioners
        </li>
      </BreadcrumbItem>
    </ol>
  </nav>
</Breadcrumb>
`;

exports[`src/pages/*/AssignPractitioners Renders correctly: Select Values 1`] = `
Array [
  Object {
    "isFixed": true,
    "label": "tlv2 - tlv2_name",
    "value": "healer",
  },
  Object {
    "isFixed": true,
    "label": "tak - Biophics Tester",
    "value": "master",
  },
  Object {
    "isFixed": true,
    "label": "lusaka - Test User Lusaka",
    "value": "437cc699-cfd7-414c-ba27-1668b6b517e6",
  },
]
`;

exports[`src/pages/*/AssignPractitioners Renders correctly: add Practitioners button 1`] = `
Array [
  <Button
    className="btn btn-primary mr-4 disabled"
    color="secondary"
    id="add-button"
    onClick={[Function]}
    tag="button"
  >
    <button
      aria-label={null}
      className="btn btn-primary mr-4 disabled btn btn-secondary"
      id="add-button"
      onClick={[Function]}
      type="button"
    >
      Save changes
    </button>
  </Button>,
  <button
    aria-label={null}
    className="btn btn-primary mr-4 disabled btn btn-secondary"
    id="add-button"
    onClick={[Function]}
    type="button"
  >
    Save changes
  </button>,
]
`;

exports[`src/pages/*/AssignPractitioners Renders correctly: add Practitioners button 2`] = `
Array [
  <Button
    className="btn btn-primary disabled"
    color="secondary"
    id="discard-button"
    onClick={[Function]}
    tag="button"
  >
    <button
      aria-label={null}
      className="btn btn-primary disabled btn btn-secondary"
      id="discard-button"
      onClick={[Function]}
      type="button"
    >
      Discard changes
    </button>
  </Button>,
  <button
    aria-label={null}
    className="btn btn-primary disabled btn btn-secondary"
    id="discard-button"
    onClick={[Function]}
    type="button"
  >
    Discard changes
  </button>,
]
`;

exports[`src/pages/*/AssignPractitioners Renders correctly: async select 1`] = `
<input
  aria-autocomplete="list"
  autoCapitalize="none"
  autoComplete="off"
  autoCorrect="off"
  disabled={false}
  id="react-select-3-input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  spellCheck="false"
  style={
    Object {
      "background": 0,
      "border": 0,
      "boxSizing": "content-box",
      "color": "inherit",
      "fontSize": "inherit",
      "label": "input",
      "opacity": 1,
      "outline": 0,
      "padding": 0,
      "width": "2px",
    }
  }
  tabIndex="0"
  type="text"
  value=""
/>
`;
